### ISSUE TITLE
wrong type for field (count) 1 != 16

### ISSUE THREAD

**Issue by cschwarz-inco** [paralect/robomongo#475]
When I open the Collections for a very large database, Robomongo logs the following error in /tmp/robomongo.log and the list of collections does not open:

```
Thu Feb 13 13:15:03.382 Assertion: 13111:wrong type for field (count) 1 != 16
0x695a56 0x68fd19 0x68fe3c 0x56321b 0x61f188 0x60c11a 0x612f79 0x613314 0x5efc7f 0x7f86d40ad009 0x7f86d40ad8bc 0x60a021 0x7f86d30af93c 0x7f86d30b5106 0x7f86d40a04c4 0x7f86d40a49b9 0x7f86d40efdc3 0x7f86d15076f2 0x7f86d150b568 0x7f86d150b71c 
 /home/cschwarz/robomongo-0.8.4-x86_64/bin/robomongo(_ZN5mongo15printStackTraceERSo+0x26) [0x695a56]
 /home/cschwarz/robomongo-0.8.4-x86_64/bin/robomongo(_ZN5mongo11msgassertedEiPKc+0xa9) [0x68fd19]
 /home/cschwarz/robomongo-0.8.4-x86_64/bin/robomongo() [0x68fe3c]
 /home/cschwarz/robomongo-0.8.4-x86_64/bin/robomongo(_ZNK5mongo11BSONElement3chkEi+0x12b) [0x56321b]
 /home/cschwarz/robomongo-0.8.4-x86_64/bin/robomongo(_ZN9Robomongo9BsonUtils6detail8getFieldIiEET_RKN5mongo11BSONElementE+0x18) [0x61f188]
 /home/cschwarz/robomongo-0.8.4-x86_64/bin/robomongo(_ZN9Robomongo19MongoCollectionInfoC1EN5mongo7BSONObjE+0x17a) [0x60c11a]
 /home/cschwarz/robomongo-0.8.4-x86_64/bin/robomongo(_ZN9Robomongo11MongoClient19runCollStatsCommandERKSs+0x499) [0x612f79]
 /home/cschwarz/robomongo-0.8.4-x86_64/bin/robomongo(_ZN9Robomongo11MongoClient19runCollStatsCommandERKSt6vectorISsSaISsEE+0xa4) [0x613314]
 /home/cschwarz/robomongo-0.8.4-x86_64/bin/robomongo(_ZN9Robomongo11MongoWorker6handleEPNS_26LoadCollectionNamesRequestE+0x7f) [0x5efc7f]
 /home/cschwarz/robomongo-0.8.4-x86_64/bin/../lib/libQt5Core.so.5(_ZNK11QMetaMethod6invokeEP7QObjectN2Qt14ConnectionTypeE22QGenericReturnArgument16QGenericArgumentS5_S5_S5_S5_S5_S5_S5_S5_S5_+0x579) [0x7f86d40ad009]
 /home/cschwarz/robomongo-0.8.4-x86_64/bin/../lib/libQt5Core.so.5(_ZN11QMetaObject12invokeMethodEP7QObjectPKcN2Qt14ConnectionTypeE22QGenericReturnArgument16QGenericArgumentS7_S7_S7_S7_S7_S7_S7_S7_S7_+0x51c) [0x7f86d40ad8bc]
 /home/cschwarz/robomongo-0.8.4-x86_64/bin/robomongo(_ZN9Robomongo18EventBusDispatcher5eventEP6QEvent+0x261) [0x60a021]
 /home/cschwarz/robomongo-0.8.4-x86_64/bin/../lib/libQt5Widgets.so.5(_ZN19QApplicationPrivate13notify_helperEP7QObjectP6QEvent+0xac) [0x7f86d30af93c]
 /home/cschwarz/robomongo-0.8.4-x86_64/bin/../lib/libQt5Widgets.so.5(_ZN12QApplication6notifyEP7QObjectP6QEvent+0x356) [0x7f86d30b5106]
 /home/cschwarz/robomongo-0.8.4-x86_64/bin/../lib/libQt5Core.so.5(_ZN16QCoreApplication14notifyInternalEP7QObjectP6QEvent+0x84) [0x7f86d40a04c4]
 /home/cschwarz/robomongo-0.8.4-x86_64/bin/../lib/libQt5Core.so.5(_ZN23QCoreApplicationPrivate16sendPostedEventsEP7QObjectiP11QThreadData+0x279) [0x7f86d40a49b9]
 /home/cschwarz/robomongo-0.8.4-x86_64/bin/../lib/libQt5Core.so.5(+0x2b8dc3) [0x7f86d40efdc3]
 /lib/libglib-2.0.so.0(g_main_context_dispatch+0x1f2) [0x7f86d15076f2]
 /lib/libglib-2.0.so.0(+0x42568) [0x7f86d150b568]
 /lib/libglib-2.0.so.0(g_main_context_iteration+0x6c) [0x7f86d150b71c]
Thu Feb 13 13:15:03.386 ERROR: [Robomongo] wrong type for field (count) 1 != 16
```

**Comment by hackpnz** [comment]
Is there some progress on this issue? I have the same problem - can't open list of collections in case of a huge DB...

**Comment by alexpts** [comment]
+1

**Comment by zoonman** [comment]
Guys, it seems I have that issue.

Cannot load list of indexes.

Error:
wrong type for field (background) 16 != 8


A little bit statistic on the collection:

```
{
    "ns" : "messages.messages",
    "count" : 6078207,
    "size" : 6772252402.0,
    "avgObjSize" : 1114,
    "storageSize" : 4061638656.0,
    "capped" : false,
    "wiredTiger" : {
        "metadata" : {
            "formatVersion" : 1
        },
        "creationString" : "allocation_size=4KB,app_metadata=(formatVersion=1),block_allocation=best,block_compressor=snappy,cache_resident=0,checkpoint=(WiredTigerCheckpoint.254253=(addr=\"--removed--\",order=254253,time=1482243469,size=4823040000,write_gen=133429490)),checkpoint_lsn=(270304,88759040),checksum=on,collator=,columns=,dictionary=0,format=btree,huffman_key=,huffman_value=,id=148,internal_item_max=0,internal_key_max=0,internal_key_truncate=,internal_page_max=4KB,key_format=q,key_gap=10,leaf_item_max=0,leaf_key_max=0,leaf_page_max=32KB,leaf_value_max=64MB,memory_page_max=10m,os_cache_dirty_max=0,os_cache_max=0,prefix_compression=0,prefix_compression_min=4,split_deepen_min_child=0,split_deepen_per_child=0,split_pct=90,value_format=u,version=(major=1,minor=1)",
        "type" : "file",
        "uri" : "statistics:table:collection-141--11346734220783974",
        "LSM" : {
            "bloom filters in the LSM tree" : 0,
            "bloom filter false positives" : 0,
            "bloom filter hits" : 0,
            "bloom filter misses" : 0,
            "bloom filter pages evicted from cache" : 0,
            "bloom filter pages read into cache" : 0,
            "total size of bloom filters" : 0,
            "sleep for LSM checkpoint throttle" : 0,
            "chunks in the LSM tree" : 0,
            "highest merge generation in the LSM tree" : 0,
            "queries that could have benefited from a Bloom filter that did not exist" : 0,
            "sleep for LSM merge throttle" : 0
        },
        "block-manager" : {
            "file allocation unit size" : 4096,
            "blocks allocated" : 134058159,
            "checkpoint size" : 4823040000.0,
            "allocations requiring file extension" : 327085,
            "blocks freed" : 132991642,
            "file magic number" : 120897,
            "file major version number" : 1,
            "minor version number" : 0,
            "file bytes available for reuse" : 279973888,
            "file size in bytes" : 4061638656.0
        },
        "btree" : {
            "btree checkpoint generation" : 254844,
            "column-store variable-size deleted values" : 0,
            "column-store fixed-size leaf pages" : 0,
            "column-store internal pages" : 0,
            "column-store variable-size leaf pages" : 0,
            "pages rewritten by compaction" : 0,
            "number of key/value pairs" : 0,
            "fixed-record size" : 0,
            "maximum tree depth" : 4,
            "maximum internal page key size" : 368,
            "maximum internal page size" : 4096,
            "maximum leaf page key size" : 2867,
            "maximum leaf page size" : 32768,
            "maximum leaf page value size" : 67108864,
            "overflow pages" : 0,
            "row-store internal pages" : 0,
            "row-store leaf pages" : 0
        },
        "cache" : {
            "bytes read into cache" : NumberLong(3718128933581),
            "bytes written from cache" : 965904276892.0,
            "checkpoint blocked page eviction" : 136,
            "unmodified pages evicted" : 171509394,
            "page split during eviction deepened the tree" : 3,
            "modified pages evicted" : 61288014,
            "data source pages selected for eviction unable to be evicted" : 136641,
            "hazard pointer blocked page eviction" : 126569,
            "internal pages evicted" : 57126,
            "pages split during eviction" : 736813,
            "in-memory page splits" : 831,
            "overflow values cached in memory" : 0,
            "pages read into cache" : 233531167,
            "overflow pages read into cache" : 0,
            "pages written from cache" : 133549569
        },
        "compression" : {
            "raw compression call failed, no additional data available" : 0,
            "raw compression call failed, additional data available" : 0,
            "raw compression call succeeded" : 0,
            "compressed pages read" : 161182032,
            "compressed pages written" : 35742315,
            "page written failed to compress" : 75974,
            "page written was too small to compress" : 97603917
        },
        "cursor" : {
            "create calls" : 24675035,
            "insert calls" : 295839289,
            "bulk-loaded cursor-insert calls" : 0,
            "cursor-insert key and value bytes inserted" : 483479161525.0,
            "next calls" : 37151533,
            "prev calls" : 1,
            "remove calls" : 87791950,
            "cursor-remove key bytes removed" : 428840937,
            "reset calls" : 1513700304.0,
            "search calls" : 1377485639.0,
            "search near calls" : 194159,
            "update calls" : 0,
            "cursor-update value bytes updated" : 0
        },
        "reconciliation" : {
            "dictionary matches" : 0,
            "internal page multi-block writes" : 471927,
            "leaf page multi-block writes" : 828602,
            "maximum blocks required for a page" : 30,
            "internal-page overflow keys" : 0,
            "leaf-page overflow keys" : 0,
            "overflow values written" : 0,
            "pages deleted" : 4706206,
            "page checksum matches" : 70608067,
            "page reconciliation calls" : 86020051,
            "page reconciliation calls for eviction" : 38018959,
            "leaf page key bytes discarded using prefix compression" : 0,
            "internal page key bytes discarded using suffix compression" : 9421135
        },
        "session" : {
            "object compaction" : 0,
            "open cursor count" : 24675013
        },
        "transaction" : {
            "update conflicts" : 1695
        }
    },
    "nindexes" : 18,
    "totalIndexSize" : 1935286272.0,
    "indexSizes" : {
        "_id_" : 431480832,
        "read-unread" : 63868928,
        "flag" : 54910976,
        "social-profile" : 30482432,
        "network_user_id" : 95297536,
        "leads" : 84262912,
        "feed" : 25821184,
        "real_time_search_id" : 88698880,
        "ttl.fast_1" : 26775552,
        "status" : 165109760,
        "ttl.regular_1" : 59830272,
        "received_date" : 155025408,
        "network" : 28258304,
        "received_date_-1_profile_id_1_feed_type_1" : 378556416,
        "profile_id_1_feed_type_1_received_date_-1" : 61808640,
        "from_1" : 101285888,
        "owner_id" : 35385344,
        "replied_idx" : 48427008
    },
    "ok" : 1.0
}
```

**Comment by Nullpo** [comment]
Same problem with Robomongo 0.9.0 in ubuntu 16.04 64 bits.
